src/generic/stage1/json_scanner.h

```diff
@@ -77,12 +116,24 @@ really_inline uint64_t follows(const uint64_t match, uint64_t &overflow) {
 
 really_inline json_block json_scanner::next(const simd::simd8x64<uint8_t>& in) {
   json_string_block strings = string_scanner.next(in);
+  // identifies the white-space and the structurat characters
   json_character_block characters = json_character_block::classify(in);
-  uint64_t follows_scalar = follows(characters.scalar(), prev_scalar);
+  // The term "scalar" refers to anything except structural characters and white space
+  // (so letters, numbers, quotes).
+  // We want  follows_scalar to mark anything that follows a non-quote scalar (so letters and numbers).
+  //
+  // A terminal quote should either be followed by a structural character (comma, brace, bracket, colon)
+  // or nothing. However, we still want ' "a string"true ' to mark the 't' of 'true' as a potential
+  // pseudo-structural character just like we would if we had  ' "a string" true '; otherwise we
+  // may need to add an extra check when parsing strings.
+  //
+  // Performance: there are many ways to skin this cat.
+  const uint64_t space_quote_structural = (characters.op() | characters.whitespace() | strings.quote());
```

Comment:
Ok. Let us do that.

Reply:
>  but I think we can make it cleaner by pushing it all the way back to the json_character_block. ~space_quote_structural is the same as nonquote_scalar, so we could replace json_character_block.scalar() with json_character_block.nonquote_scalar() and call that where we currently call scalar().

But we can't because json_character_block does not know about quotes.

Reply:
I can code it like this though...

```C++
  const uint64_t nonquote_scalar = characters.scalar() & ~strings.quote();
  uint64_t follows_nonquote_scalar = follows(nonquote_scalar, prev_scalar);
```

Reply:
I pushed an update. Let us see what the tests are saying.